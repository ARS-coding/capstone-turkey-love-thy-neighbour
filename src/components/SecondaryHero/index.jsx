import React from "react";

import { useTranslation } from "react-i18next";

import { Container, Row, Col } from "react-bootstrap";

import img from "../../images/SecondaryHeroContentBgImg.png";
import "./index.scss";

function SecondaryHero() {
  const { t } = useTranslation();
  return (
    <section className="snapping-section secondary-hero-img-bg container-fluid">
      <Container className="secondary-hero-content-container d-flex align-items-center">
        <Row className="secondary-hero-content-row m-0">
          <Col
            xs={{ span: 12, order: "last" }}
            sm={{ span: 12, order: "last" }}
            md={{ span: 6, order: "first" }}
            className="p-0 px-xs-1 px-md-1 h-100 d-flex justify-content-xs-start justify-content-sm-start justify-content-md-center align-items-md-center"
          >
            <img src={img} alt="img" className="img-fluid" />
          </Col>
          <Col
            xs={{ span: 12, order: "first" }}
            sm={{ span: 12, order: "first" }}
            md={{ span: 6, order: "last" }}
            className="px-0 pt-xs-5 pt-xs-5 pt-md-0 d-flex flex-column justify-content-center align-items-start"
          >
            <h2>{t("home_second_slogan_big")}</h2>
            <h1>{t("home_second_slogan_small")}</h1>
            <p>{t("home_lesser_slogan")}</p>
          </Col>
        </Row>
      </Container>
    </section>
  );
}

export default SecondaryHero;
