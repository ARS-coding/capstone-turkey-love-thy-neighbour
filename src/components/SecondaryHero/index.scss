@import "../../style/variables";

.secondary-hero-img-bg {
  background-image: url("../../images/SecondaryHeroBgImg.png");
  .secondary-hero-content-container {
    .secondary-hero-content-row {
      & > div:last-child {
        // background-color: red !important;
        h1 {
          color: $logo-blue;
          font-size: max(40px);
          font-weight: bold;
          font-family: Roboto, sans-serif;

          @media screen and (min-width: 768px) and (max-width: 991px) {
            font-size: 35px;
          }
          @media screen and (max-width: 371px) {
            font-size: 32px;
          }
        }

        h2 {
          color: $logo-red;
          position: relative;
          top: 10px;
          font-style: italic;
          font-weight: bold;
          font-family: Roboto, sans-serif;

          @media screen and (min-width: 768px) and (max-width: 991px) {
            font-size: 22px;
          }
          @media screen and (max-width: 371px) {
            font-size: 20px;
          }
        }

        p {
          color: white;
          font-size: 20px;
          @media screen and (min-width: 768px) and (max-width: 991px) {
            font-size: 18px;
          }
          @media screen and (max-width: 371px) {
            font-size: 17px;
          }
        }
      }
      & > div:first-child {
        img {
          width: 270px;
          max-height: 350px;
          @media screen and (min-width: 282px) and (max-width: 767px) {
            max-height: 350px;
          }
          @media screen and (max-width: 281px) {
            max-height: 330px;
          }
        }
      }
    }
  }
}
