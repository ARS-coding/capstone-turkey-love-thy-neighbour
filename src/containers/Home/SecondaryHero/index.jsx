import React from "react";

import { Container, Row, Col } from "react-bootstrap";

import img from "../../../utils/SecondaryHeroContentBgImg.png";
import "./index.scss";

function SecondaryHero() {
  return (
    <section className="secondary-hero-img-bg container-fluid">
      <Container className="secondary-hero-content-container d-flex align-items-center">
        <Row className="secondary-hero-content-row m-0">
          <Col
            xs={{ span: 12, order: "last" }}
            sm={{ span: 12, order: "last" }}
            md={{ span: 6, order: "first" }}
            className="p-0 px-xs-1 px-md-1 h-100 d-flex justify-content-xs-start justify-content-sm-start justify-content-md-center align-items-md-center"
          >
            <img src={img} alt="img" className="img-fluid" />
          </Col>
          <Col
            xs={{ span: 12, order: "first" }}
            sm={{ span: 12, order: "first" }}
            md={{ span: 6, order: "last" }}
            className="px-0 pt-xs-5 pt-xs-5 pt-md-0 d-flex flex-column justify-content-center align-items-start"
          >
            <h2>Welcome,</h2>
            <h1 className="to-be-welcomed">to be welcomed.</h1>
            <p>
              Respect each other and feel the genuine <span>connection.</span>
            </p>
          </Col>
        </Row>
      </Container>
    </section>
  );
}

export default SecondaryHero;
