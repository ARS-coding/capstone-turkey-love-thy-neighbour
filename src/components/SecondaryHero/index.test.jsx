import React from "react";
import renderer from "react-test-renderer";
import SecondaryHero from ".";

describe("SecondaryHero Page", () =>
  it("renders SecondaryHero Page correctly", () => {
    const tree = renderer.create(<SecondaryHero />).toJSON();
    expect(tree).toMatchSnapshot();
  }));
