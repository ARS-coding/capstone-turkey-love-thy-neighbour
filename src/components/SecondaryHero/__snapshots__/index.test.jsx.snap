// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecondaryHero Page renders SecondaryHero Page correctly 1`] = `
<section
  className="snapping-section secondary-hero-img-bg container-fluid"
>
  <div
    className="secondary-hero-content-container d-flex align-items-center container"
  >
    <div
      className="secondary-hero-content-row m-0 row"
    >
      <div
        className="p-0 px-xs-1 px-md-1 h-100 d-flex justify-content-xs-start justify-content-sm-start justify-content-md-center align-items-md-center col-md-6 col-sm-12 col-12 order-md-first order-sm-last order-last"
      >
        <img
          alt="img"
          className="img-fluid"
          src="SecondaryHeroContentBgImg.png"
        />
      </div>
      <div
        className="px-0 pt-xs-5 pt-xs-5 pt-md-0 d-flex flex-column justify-content-center align-items-start col-md-6 col-sm-12 col-12 order-md-last order-sm-first order-first"
      >
        <h2>
          Welcome,
        </h2>
        <h1>
          to be welcomed.
        </h1>
        <p>
          Respect each other and feel the genuine 
          <span>
            connection.
          </span>
        </p>
      </div>
    </div>
  </div>
</section>
`;
